go/vt/dbconfigs/dbconfigs.go

```diff
@@ -94,7 +94,7 @@ func registerBaseFlags() {
 	flag.StringVar(&baseConfig.UnixSocket, "db_socket", "", "The unix socket to connect on. If this is specifed, host and port will not be used.")
 	flag.StringVar(&baseConfig.Host, "db_host", "", "The host name for the tcp connection.")
 	flag.IntVar(&baseConfig.Port, "db_port", 0, "tcp port")
-	flag.StringVar(&baseConfig.Charset, "db_charset", "utf8", "Character set. Only utf8 or latin1 based character sets are supported.")
+	flag.StringVar(&baseConfig.Charset, "db_charset", "", "Character set. Only utf8 or latin1 based character sets are supported.")
```

Comment:
This seems like a good fix that is unlikely to introduce a regression from how it originally was in here: https://github.com/vitessio/vitess/blob/ecf36cecb1fafdcb6f745cfd7f12672f11873618/go/vt/dbconfigs/dbconfigs.go#L245

I believe the only difference now is:
1. if `HasConnectionParams()` is `true`, and someone was setting non-`utf8` user specific charsets, but not setting a base config charset (i.e. base config charset was defaulting to `utf8`), then previously their user charsets were always overrided by the base config. Now the user specific charsets they set will take precedence if they did not set a base config charset. That's probably what they wanted in the first place.
2. if `HasConnectionParams()` is `false` previously the base config charset was never getting used. Only the user specific charsets would get used. Now, the base config charset will override the user specific charsets if it is set. Again, this is probably what people would have wanted in the first place.